pandas/core/window/rolling.py

```diff
@@ -2102,12 +2103,101 @@ class RollingGroupby(WindowGroupByMixin, Rolling):
     Provide a rolling groupby implementation.
     """
 
+    def _apply(
+        self,
+        func: Callable,
+        center: bool,
+        require_min_periods: int = 0,
+        floor: int = 1,
+        is_weighted: bool = False,
+        name: Optional[str] = None,
+        use_numba_cache: bool = False,
+        **kwargs,
+    ):
+        result = Rolling._apply(
+            self,
+            func,
+            center,
+            require_min_periods,
+            floor,
+            is_weighted,
+            name,
+            use_numba_cache,
+            **kwargs,
+        )
+        # _wrap_outputs does not know about what the result index should be
+        grouped_object_index = self._groupby._selected_obj.index
```

Comment:
hmm, is it not possible to use self._groupby.grouper.result_index

Reply:
Unfortunately it's not exactly what I need. Since we always need to construct a MultiIndex, it's note entirely useful in this situation

```
In [1]:         df = pd.DataFrame({"A": ["foo"] * 3 + ["bar"] * 3, "B": [1] * 6})
   ...:         g = df.groupby("A")

In [2]: g.rolling(window=2).sum()
/pandas/core/window/rolling.py(2128)_apply()
-> grouped_object_index = self._groupby._selected_obj.index
(Pdb) c
/pandas/core/window/rolling.py(2147)_apply()
-> return result
(Pdb) self._groupby.grouper.result_index
Index(['bar', 'foo'], dtype='object', name='A')
(Pdb) result_index
MultiIndex([('bar', 3),
            ('bar', 4),
            ('bar', 5),
            ('foo', 0),
            ('foo', 1),
            ('foo', 2)],
           names=['A', None])
```